Animation fixes

maribelferreira opened the pull request.

maribelferreira committed:
Refactor BorderAnimation CSS with CSS variables

Improve maintainability by introducing CSS variables for border properties, colors, and animation settings. Standardize border widths and colors using semantic variable names.

maribelferreira committed:
Adjust border styling in verification animation components

Increase border width from 2px to 3px across ChainVerificationAnimation components and remove redundant background color in BorderAnimation CSS

maribelferreira committed:
Add custom animation duration for verification phases

Introduce dynamic animation timing for different verification steps:
- Add custom CSS variable for animation duration
- Conditionally set animation speed based on phase title
- Import BorderAnimation.css for styling
- Enhance active phase border with fill animation

gnarea commented on src/components/ChainVerificationAnimation/BorderAnimation.css:
```diff
@@ -1,29 +1,38 @@
+/* CSS Variables */
+:root {
+  --border-width-base: 3px;
+  --border-width-large: 3px;
```
What's the point of these two separate variables? Did you mean to make the base 2px?

gnarea commented on src/components/ChainVerificationAnimation/Phase.tsx:
```diff
@@ -34,15 +35,32 @@ export const PhaseColumn: React.FC<PhaseProps> = ({
     return isCompleted ? "border-green-500" : "border-amber-500";
   };
 
+  // Custom animation duration for DNSSEC Chain
+  const customStyle = isActive
+    ? ({
+        "--animation-duration":
+          title === "DNSSEC Chain"
+            ? "6s"
+            : title === "X.509 Certificate Chain"
+              ? "2s"
+              : title === "CMS SignedData"
+                ? "1s"
+                : "1s",
+      } as React.CSSProperties)
+    : {};
```
I think this is a really brittle way to do that. It'd break as soon as we change one of those titles, or change the number of the steps in a phase.

Why don't just dynamically compute the duration based on the number of steps? 1 step = 1 second, 2 steps = 2 seconds, etc.

maribelferreira replied:
That's a good point! Sorry, my bad.

gnarea reviewed (changes requested):
Looking great, but I have some suggestions.

maribelferreira committed:
Simplify animation duration and border styling

Refactor animation and border CSS to use more consistent and dynamic approach:
- Replace multiple border width variables with single `--border-width`
- Remove media query-specific border width adjustments
- Calculate animation duration dynamically based on step count
- Simplify border fill animation styling

maribelferreira committed:
Remove redundant CSS comment in BorderAnimation styles

gnarea commented on src/components/ChainVerificationAnimation/Phase.tsx:
```diff
@@ -35,21 +35,13 @@ export const PhaseColumn: React.FC<PhaseProps> = ({
-              : title === "CMS SignedData"
-                ? "1s"
-                : "1s",
+        "--animation-duration": `${Math.max(1, steps.length)}s`,
```
This should be wrongly capping the animation duration at 1 second, but it isn't, so I wonder if this code is actually used?

maribelferreira committed:
Remove dynamic animation duration styling from Phase component

Simplify Phase component by removing custom animation duration calculation and inline styling. This change maintains the existing border and styling logic while removing unnecessary dynamic style generation.
